tasks/mutations/bed_mutations.ts

```diff
@@ -29,4 +39,107 @@ export type BedMinimalDTO = {
   index: number
 }
 
+export type BedWithRoomID = {
+  id: string,
+  roomID: string
+}
+
 export const bedQueryKey = 'beds'
+
+export const useBedQuery = (bedID: string | undefined) => {
+  return useQuery({
+    queryKey: [bedQueryKey],
+    enabled: !!bedID,
+    queryFn: async () => {
+      const req = new GetBedRequest()
+      if (bedID) {
+        req.setId(bedID)
+      }
```

Comment:
A ID is required by the API. We should early return when `bedID` is falsy

Reply:
The query isn't excited when the bed Id isn't defined, see the enabled attribute. The if only occurs because Typescript doesn't see that.